src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -8616,6 +8614,42 @@ namespace SurfaceGeometry {
         }
     }
 
+    void CheckWindowShadingControlSimilarForWindow(bool& ErrorsFound)
+    {
+        // For each window check if all window shading controls on list are the same except for name, schedule name, construction, and material
+        for (auto theSurf : Surface) {
+            if (theSurf.HasShadeControl) {
+                if (theSurf.windowShadingControlList.size() > 1) {
+                    int firstWindowShadingControl = theSurf.windowShadingControlList.front();
+                    for (auto wsc = std::next(theSurf.windowShadingControlList.begin()); wsc != theSurf.windowShadingControlList.end(); ++wsc) {
+                        if (!isWindowShadingControlSimilar(firstWindowShadingControl, *wsc)) {
+                            ErrorsFound = true;
+                            ShowSevereError("CheckWindowShadingControlSimilarForWindow: Fenestration surface named \"" + theSurf.Name +
+                                        "\" has multiple WindowShadingContols that are not similar.");
+                            ShowContinueError("for: \"" + WindowShadingControl(firstWindowShadingControl).Name + " and: " + WindowShadingControl(*wsc).Name); 
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    bool isWindowShadingControlSimilar(int a, int b)
+    {
+        // Compares two window shading controls are the same except for the name, schedule name, construction, and material
```

Comment:
Oh, I see, this is still enforcing all of the same constraints. I guess we can work on relaxing that after v9.4.